plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/projection/DateProjectionFactory.java

```diff
@@ -0,0 +1,202 @@
+    private static final Pattern DATE_RANGE_BOUND_EXPRESSION_PATTERN = Pattern.compile("\\s*NOW\\s*(([+-])\\s*([0-9]+)\\s*(DAY|HOUR|MINUTE|SECOND)S?\\s*)?");
+
+    @Override
+    public boolean isSupportedColumnType(Type columnType)
+    {
+        return columnType instanceof VarcharType
+                || columnType instanceof DateType
+                || (columnType instanceof TimestampType && ((TimestampType) columnType).isShort());
+    }
+
+    @Override
+    public Projection create(String columnName, Type columnType, Map<String, Object> columnProperties)
+    {
+        String dateFormatPattern = getProjectionPropertyRequiredValue(
+                columnName,
+                columnProperties,
+                COLUMN_PROJECTION_FORMAT,
+                String::valueOf);
+
+        List<String> range = getProjectionPropertyRequiredValue(
+                columnName,
+                columnProperties,
+                COLUMN_PROJECTION_RANGE,
+                value -> ((List<?>) value).stream()
+                        .map(String.class::cast)
+                        .collect(toImmutableList()));
+        if (range.size() != 2) {
+            throw invalidRangeProperty(columnName, dateFormatPattern);
+        }
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat(dateFormatPattern);
+        dateFormat.setLenient(false);
+        dateFormat.setTimeZone(TimeZone.getTimeZone(UTC_TIME_ZONE_ID));
+        Supplier<Instant> leftBound = parseDateRangerBound(columnName, range.get(0), dateFormat);
+        Supplier<Instant> rangeBound = parseDateRangerBound(columnName, range.get(1), dateFormat);
+        if (!leftBound.get().isBefore(rangeBound.get())) {
+            throw invalidRangeProperty(columnName, dateFormatPattern);
+        }
+
+        int interval = getProjectionPropertyValue(columnProperties, COLUMN_PROJECTION_INTERVAL, Integer.class::cast).orElse(1);
+        ChronoUnit intervalUnit = getProjectionPropertyValue(columnProperties, COLUMN_PROJECTION_INTERVAL_UNIT, ChronoUnit.class::cast)
+                .orElseGet(() -> resolveDefaultChronoUnit(columnName, dateFormatPattern));
+
+        if (!DATE_PROJECTION_INTERVAL_UNITS.contains(intervalUnit)) {
+            throw invalidProjectionException(
+                    columnName,
+                    format(
+                            "Property: '%s' value '%s' is invalid. Available options: %s",
+                            COLUMN_PROJECTION_INTERVAL_UNIT,
+                            intervalUnit,
+                            DATE_PROJECTION_INTERVAL_UNITS));
+        }
+
+        return new DateProjection(columnName, dateFormat, leftBound, rangeBound, interval, intervalUnit);
+    }
+
+    private ChronoUnit resolveDefaultChronoUnit(String columnName, String dateFormatPattern)
+    {
+        if (dateFormatPattern.contains("S") || dateFormatPattern.contains("s")
```

Comment:
what about my comment that there can be arbitrary, uninterpreted characters in the pattern (if we allow anything `SimpleDateFormat` allows)

E.g. sth like `"dd'days'"` will now be rejected - even though it should return `DAYS`.

Reply:
Maybe we should just reject if `'` is in the pattern?

Reply:
Sorry I cant recall such comment. I've must've missed it. 
Hmm this is valid point, i will have to thing it over how to approach it.

Reply:
I can imagine partitions like 20200101T101010 which would required 'T' inside :/

Reply:
Ok will sanitise it from text like this `dateFormatPattern.replaceAll("'.*'", "")`

Reply:
Also added test case for it.